runtime/j9vm/jvm.c

```diff
@@ -4474,11 +4504,18 @@ JVM_RaiseSignal(jint sigNum)
 {
 	jboolean rc = JNI_FALSE;
 	J9JavaVM *javaVM = (J9JavaVM *)BFUjavaVM;
+	BOOLEAN isShutdownSignal = isSignalUsedForShutdown(sigNum);
 
 	Trc_SC_RaiseSignal_Entry(sigNum);
 
-	if (J9_ARE_ALL_BITS_SET(javaVM->sigFlags, J9_SIG_XRS)
-			&& isSignalSpecial(sigNum)) {
+	if ((J9_ARE_ANY_BITS_SET(javaVM->sigFlags, J9_SIG_XRS | J9_SIG_XRS_ASYNC)
+			&& isShutdownSignal)
+#if defined(SIGQUIT)
+			|| (SIGQUIT == sigNum)
+#endif /* defined(SIGQUIT) */
+	) {
+		/* Ignore signal */
+	} else if (isShutdownSignal && isSignalIgnored(sigNum)) {
```

Comment:
Why only check for shutdown signals?

Reply:
this requirement was part of JCL team's specification